test/concurrent/ThreadTest.ooc

```diff
@@ -45,15 +45,17 @@ ThreadTest: class extends Fixture {
 				value set(value get() + 1)
 				Thread yield()
 				Time sleepMilli(1)
+				if (Thread currentThread() isCanceled())
```

Comment:
We've got a problem here. `currentThread()` is
```ooc
thread := This new(func)
thread pthread = pthread_self()
thread
```
meaning it, first of all, creates a new `ThreadUnix` instance every time (which must be freed). That's OK.

But more importantly here, this new instance will not have the *same* `cancelMutex` or `canceled` members, and so checking if it's canceled will always return false... :( I can solve it by adding some reference variable which, if not null, propagates the `isCanceled` call to that thread instance. That's not too pretty though. Any other idea? @sebastianbaginski